### ISSUE TITLE
Commit 5db781cc3d47df639a132533de67769c4f5d73e8

### ISSUE THREAD

**Issue by dhalsimax** [libsdl-org/SDL#11031]
On this commit lines:

```c
SDL_RenderDriver GLES2_RenderDriver = {
    GLES2_CreateRenderer,
    { "opengles2",
      (SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC | SDL_RENDERER_TARGETTEXTURE),
      4,
      { SDL_PIXELFORMAT_RGBA32,
        SDL_PIXELFORMAT_BGRA32,
        SDL_PIXELFORMAT_BGRX32,
        SDL_PIXELFORMAT_RGBX32 },
      0,
      0 }
};
```

shoudl be:

```c
SDL_RenderDriver GLES2_RenderDriver = {
    GLES2_CreateRenderer,
    { "opengles2",
      (SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC | SDL_RENDERER_TARGETTEXTURE),
      4,
      { 
        SDL_PIXELFORMAT_BGRA32,
        SDL_PIXELFORMAT_RGBA32,
        SDL_PIXELFORMAT_BGRX32,
        SDL_PIXELFORMAT_RGBX32 },
      0,
      0 }
};
```

Thus because many of us still get the first item of the list as default for creating textures as targets or for direct fill.

**Comment by icculus** [comment]
5db781cc3d47df639a132533de67769c4f5d73e8 changed this to support bigendian systems, but yeah, I think this specific piece might have been incorrect to change.

**Comment by dhalsimax** [comment]
> [5db781c](https://github.com/libsdl-org/SDL/commit/5db781cc3d47df639a132533de67769c4f5d73e8) changed this to support bigendian systems, but yeah, I think this specific piece might have been incorrect to change.

I think it was a simple oversight, the irst item array was "SDL_PIXELFORMAT_ARGB8888" and not "SDL_PIXELFORMAT_ABGR8888" so for correctness and analogy the commit should be: "SDL_PIXELFORMAT_ARGB8888->SDL_PIXELFORMAT_RBRA32" and so the second substitution "SDL_PIXELFORMAT_ABGR8888->SDL_PIXELFORMAT_RGBA32". Items 3 and 4 substitution are instead both correct.

**Comment by icculus** [comment]
Should be fixed with 9f7ccbe8d406b9605acb2cdbdc63192f9646603d, let me know if it isn't!

**Comment by ccawley2011** [comment]
For reference, the reason for having `SDL_PIXELFORMAT_RGBA32` at the top of the list is because corresponds to the format natively used by OpenGL ES (without having to swap the R and B components using pixel shaders), which makes it more fitting as a default format for applications that select the pixel format dynamically to avoid the extra format conversion.

Is there a particular benefit to having `SDL_PIXELFORMAT_BGRA32` as the default for OpenGL ES 2 instead?
